### PR TITLE
Adding support for `NSCopying` protocol.

### PR DESCRIPTION
**Pull request by JThramer**
What it says on the box. Making `RouteObject` conform to `NSCopying`. Needed for mapbox/mapbox-navigation-ios#806.

/cc @1ec5 @bsudekum

### CODE REVIEW

File: MapboxDirections/MBRouteOptions.swift [hunk 1]
```diff
@@ -458,6 +458,26 @@ open class RouteOptions: NSObject, NSSecureCoding {
         }
         return (waypoints, routes)
     }
+    
+    // MARK: NSCopying
+    /**
+     Returns a copy of the object.
+     - parameter zone: an NSZone to use in the copy operation.
+     - returns: A copy of the object.
```
Comment (1ec5) [inline MapboxDirections/MBRouteOptions.swift]: I _think_ this method already inherits documentation from the protocol’s method declaration, at least as far as Quick Help and other tools are concerned.

**Comment by frederoni** [comment]
Would it be possible to avoid all this boilerplate code by adopting Codable now that we're on an Xcode 9 stack already?

**Comment by JThramer** [comment]
I agree @frederoni, but wouldn't it be better to do that refactor in one tech-debt chore, rather than piecemeal? There are 8 classes that use `NSSecureCoding`, and two that use `NSCopying`, so it seems like there's enough of a payload to justify a chore here.

File: MapboxDirections/MBRouteOptions.swift [hunk 2]
```diff
@@ -476,8 +477,32 @@ open class RouteOptions: NSObject, NSSecureCoding, NSCopying{
...
     }
+    
+    //MARK: - OBJ-C Equality
+    open override func isEqual(_ object: Any?) -> Bool {
+        guard let opts = object as? RouteOptions else { return false }
+        return isEqual(to: opts)
+    }
+    
+    @objc(isEqualToRouteOptions:)
+    open func isEqual(to routeOptions: RouteOptions?) -> Bool {
```
Comment (1ec5) [inline MapboxDirections/MBRouteOptions.swift]: Cool, this gets us a little bit closer to implementing #151. 👍

**Pull request merged**